src/datasets/summarizations/category-topk.summarization.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { queryFactory } from './category-topk.summarization';
+import { getSummaries, hasHighValidity, isTextPartsInSummary } from './utils/tests';
+import { CategoricalPoint } from '../metas/types';
+
+describe('queryFactory', () => {
+
+  const points: CategoricalPoint[] = [
+    { x: 'A', y: 100, },
+    { x: 'B', y: 200, },
+    { x: 'C', y: 300, },
+    { x: 'D', y: 400, },
+  ];
+
+  it('should have validity value between 0 and 1.', () => {
+    const summaries = getSummaries(queryFactory, points);
+    for (const summary of summaries) {
+      expect(summary.validity).toBeGreaterThanOrEqual(0.0);
+      expect(summary.validity).toBeLessThanOrEqual(1.0);
+    }
+  });
+
+  it('should describe top 3 categories by default.', () => {
+    const summaries = getSummaries(queryFactory, points);
+    const isTop1stSummary = isTextPartsInSummary('D', '1st', 'sessions');

```

Comment:
nit: can't find our old thread, but I still think it would be nice to test the full templates.